external-import/doppel/src/external_import_connector/converter_to_stix.py

```diff
@@ -0,0 +1,109 @@
+import json
+
+from pycti import Identity as PyCTIIdentity
+from pycti import Indicator as PyCTIIndicator
+from stix2 import Bundle, Identity, Indicator
+
+from .utils import parse_iso_datetime
+
+
+class ConverterToStix:
+    def __init__(self, helper, config):
+        self.helper = helper
+        self.config = config
+        self.author = self._create_identity()
+
+    def _create_identity(self):
+        return Identity(
+            id=PyCTIIdentity.generate_id(name="Doppel", identity_class="organization"),
+            name="Doppel",
+            identity_class="organization",
+            description="Threat Intelligence Provider",
+            allow_custom=True,
+        )
+
+    def convert_alerts_to_stix(self, alerts):
+        stix_objects = [self.author]
+        created_by_ref = self.author.id
+
+        for alert in alerts:
+            entity = alert.get("entity", "Unknown")
+            pattern = f"[entity:value ='{entity}']"
+            alert_id = alert.get("id", "unknown")
+
+            # Corrected: Removed invalid keyword argument `pattern_type`
+            indicator_id = PyCTIIndicator.generate_id(pattern=pattern)
+
+            self.helper.log_info(f"Processing alert ID: {alert_id}")
+
+            created_at = parse_iso_datetime(
+                alert.get("created_at", ""), "created_at", alert_id, self.helper
+            )
+            modified = parse_iso_datetime(
+                alert.get("last_activity_timestamp", ""),
+                "last_activity_timestamp",
+                alert_id,
+                self.helper,
+            )
+
+            audit_logs = alert.get("audit_logs", [])
+            audit_log_text = "\n".join(
+                [
+                    f"{log['timestamp']}: {log['type']} - {log['value']}"
+                    for log in audit_logs
+                ]
+            )
+
+            entity_content = alert.get("entity_content", {})
+            formatted_entity_content = json.dumps(entity_content, indent=2)
+            platform = alert.get("platform", "unknown")
+            platform_value = alert.get("product", "Unknown")
+
+            entity_state = alert.get("entity_state", "unknown")
+            queue_state = alert.get("queue_state", "unknown")
+            raw_severity = alert.get("severity", "unknown")
+            severity = f"{raw_severity} severity"
+
+            description = (
+                f"Platform: {platform},\n"
+                f"Entity State: {entity_state},\n"
+                f"Queue State: {queue_state},\n"
+                f"Severity: {severity},\n"
+                f"Entity Content:\n{formatted_entity_content}"
+            )
```

Comment:
The description field uses Markdown formatting if you want. 
I added :
- the syntax `  \n` instead of `,\n` to make sure the new line is present on OpenCTI (cf: [source](https://www.markdownguide.org/basic-syntax/#line-break-best-practices))
- the double `**` to bold the titles as an example)

This syntax will display it properly : 
```suggestion
            description = (
                 f"**Platform**: {platform}  \n"
                 f"**Entity State**: {entity_state}  \n"
                 f"**Queue State**: {queue_state}  \n"
                 f"**Severity**: {severity}  \n"
                 f"**Entity Content**:  \n
                 f"{formatted_entity_content}"
             )
```